content/en/tracing/manual_instrumentation/java.md

```diff
@@ -1,36 +1,488 @@
+
+The `dd.tags` property allows setting tags across all generated spans for an application. This can be useful for grouping stats for your applications, datacenters, or any other tags you would like to see within the Datadog UI.
+
+```text
+java -javaagent:<DD-JAVA-AGENT-PATH>.jar \
+     -Ddd.tags=datacenter:njc,<TAG_KEY>:<TAG_VALUE> \
+     -jar <YOUR_APPLICATION_PATH>.jar
+```
+
+
+
+### Set errors on a span
+
+To customize an error associated with one of your spans, set the error tag on the span and use `Span.log()` to set an “error event”.  The error event is a `Map<String,Object>` containing a `Fields.ERROR_OBJECT->Throwable` entry, a `Fields.MESSAGE->String`, or both.
 
-If you aren't using a [supported framework instrumentation][1], or you would like additional depth in your application’s [traces][2], you may want to manually instrument your code.
+```java
+import io.opentracing.Span;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.GlobalTracer;
+import io.opentracing.log.Fields
+...
+   // Get active span if not available in current method
+    final Span span = GlobalTracer.get().activeSpan();
+    if (span != null) {
+      span.setTag(Tags.ERROR, true);
+      span.log(Collections.singletonMap(Fields.ERROR_OBJECT, ex));
+    }
+```
 
-Do this either using the Trace annotation for simple method call tracing, or with the OpenTracing API for complex tracing. Datadog's Trace annotation is provided by the [dd-trace-api dependency][3]
+Note: `Span.log()` is a generic OpenTracing mechanism for associating events to the current timestamp.  The Java Tracer only supports logging error events.
+Alternatively, you can set error tags directly on the span without `log()`:
+
+```java
+import io.opentracing.Span;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.GlobalTracer;
+import datadog.trace.api.DDTags;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+...
+   final Span span = GlobalTracer.get().activeSpan();
+    if (span != null) {
+      span.setTag(Tags.ERROR, true);
+      span.setTag(DDTags.ERROR_MSG, ex.getMessage());
+      span.setTag(DDTags.ERROR_TYPE, ex.getClass().getName());
+
+      final StringWriter errorString = new StringWriter();
+      ex.printStackTrace(new PrintWriter(errorString));
+      span.setTag(DDTags.ERROR_STACK, errorString.toString());
+    }
+```
+
+Note: Any relevant error metadata explained in the Trace View docs can also be added. If the current span isn’t the root span, mark it as an error by using the dd-trace-api library to grab the root span with `MutableSpan`, then use `setError(true)`. See the [setting tags & errors on a root span][9] section for more details.
+
+
+
+### Set tags & errors on a root span from a child span
+
+When an event or condition happens downstream, you may want that behavior or value reflected as a tag on the top level or root span.  This can be useful for count an error for measuring  performance, or setting a dynamic tag for observability.
```

Comment:
```suggestion
When an event or condition happens downstream, you may want that behavior or value reflected as a tag on the top level or root span. This can be useful to count an error or for measuring performance, or setting a dynamic tag for observability.
```